### PR TITLE
Remember the theme and service throughout the password reset flow, including email

### PR DESCRIPTION
**Pull request by kevin-wise-infusionsoft**
Remember the theme and service throughout the various flows, including password reset and email

- Also: on reset form, remember username from login form
- Fixed password-related js paths

### CODE REVIEW

**Comment by dkpalmer** [comment]
@bbooth I believe CAS made it into prod? Any objections to moving this one through?

/cc @joshuajohnson814

**Comment by bbooth** [comment]
@dkpalmer  I want to hold off until wed.  We will be ramping up load on CAS for the next few days.  If anything comes up I want to be able to fix it without doing crazy branching if possible.

**Comment by dkpalmer** [comment]
Got it. Thanks.

**Comment by bbooth** [comment]
Technically we it not in prod yet :)  Hopefully in the next 30 minutes.

**Comment by dkpalmer** [comment]
giphy

**Comment by dkpalmer** [comment]
Thoughts on this one @bbooth ?

**Comment by bbooth** [comment]
I am going to go through this again over the weekend.  I will merge it after if it is all good.

**Comment by bbooth** [comment]
Been testing this today.  The password recovery for the new theme is messed up.  Not sure if this is from the other changes Chris did that I merged earlier today.

image

**Comment by dkpalmer** [comment]
@bbooth Is that independent of this PR where we can address that separately?

File: cas/src/main/java/org/apereo/cas/infusionsoft/web/controllers/RegistrationController.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
  * Controller that backs the new user registration and "forgot password" flows.
  */
 @Controller
-@RequestMapping(value = {"/registration", "/app/registration"})
+@RequestMapping(value = {"/registration", "/app/registration", "/password"})
```
Comment (bbooth) [inline cas/src/main/java/org/apereo/cas/infusionsoft/web/controllers/RegistrationController.java]: Why are we adding the /password mapping here?

File: cas/src/main/java/org/apereo/cas/infusionsoft/web/controllers/RegistrationController.java [hunk 2]
```diff
@@ -439,11 +442,16 @@ public String reset(Model model, String recoveryCode, String password1, String p
 
             return "registration/reset";
         } else {
-            if (user != null) {
-                autoLoginService.autoLogin(user.getUsername(), request, response);
-            }
+            autoLoginService.autoLogin(user.getUsername(), request, response);
 
-            return "redirect:" + defaultRedirectUrl;
+            // Redirect directly to the login URL, which will redirect to the redirectUrl with a service ticket
```
Comment (bbooth) [inline cas/src/main/java/org/apereo/cas/infusionsoft/web/controllers/RegistrationController.java]: Seems that autologin isn't working for some reason.  After reseting the password the user is required to login again.  I dig to see why but something is not correct.

**Comment by bbooth** [comment]
Not sure how to test this exactly, can I get some sample requests.

**Comment by bbooth** [comment]
I think the password reset should be addressed because the current experience is the old password recovery theme which looks fine.  This introduces something new even though it is possible that it was just uncovered by these changes.

**Comment by joshuajohnson814** [comment]
@bbooth, @dkpalmer -> @kevin-wise-infusionsoft is out for another week. Can you help us out with this?

**Comment by bbooth** [comment]
I am going to rebase this on master and see if the css and weird autologin stuff I was seeing goes away

**Comment by bbooth** [comment]
I rebased and it seems to have taken care of the autologin wierdness.

**Comment by dkpalmer** [comment]
Nice. What's left after that.. just the CSS?

**Comment by joshuajohnson814** [comment]
@bbooth @dkpalmer can we get this merged and follow up with CSS fixes when @kevin-wise-infusionsoft gets back?

**Comment by bbooth** [comment]
css fixes look good.

**Pull request merged**